### PR TITLE
To add local registry to libvirt_coreos

### PR DESCRIPTION
**Pull request by sdminonne**
`libvirt_coreos` is quick way to have a multi-node cluster on a linux laptop for development purpouse.
This PR adds local registry to libvirt_coreos cluster. Mind have a look?

@roberthbailey since you reviewed my last PR on dns for libvirt_coreos

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36751)

### CODE REVIEW

**Comment by roberthbailey** [comment]
@sdminonne - this change lgtm although we should be clear that we want to require all new libvirt clusters to run with a local registry. I assume that since the primary purpose is local development that this is a good thing, but it might be worth adding an option to disable it if you want to run a leaner cluster footprint on your laptop.

**Comment by sdminonne** [comment]
@roberthbailey thanks. Going to add an ENV var to switch on/off local registry.

**Comment by sdminonne** [comment]
@roberthbailey added conditional creation via`KUBE_ENABLE_CLUSTER_REGISTRY` (default `false`) and squashed (since it's not a big change).
PTAL

File: cluster/libvirt-coreos/config-default.sh [hunk 1]
```diff
@@ -58,9 +58,16 @@ ENABLE_CLUSTER_DNS="${KUBE_ENABLE_CLUSTER_DNS:-true}"
 DNS_SERVER_IP="${SERVICE_CLUSTER_IP_RANGE%.*}.254"
 DNS_DOMAIN="cluster.local"
 
+# Optional: Install cluster registry
+ENABLE_CLUSTER_REGISTRY="${KUBE_ENABLE_CLUSTER_REGISTRY:-false}"
```
Comment (roberthbailey) [inline cluster/libvirt-coreos/config-default.sh]: I think we should default this to true but give folks a way to opt-out from a local registry.

File: cluster/libvirt-coreos/util.sh [hunk 2]
```diff
@@ -279,6 +285,51 @@ function kube-up {
...
+
+
+function create_registry_svc() {
+  echo " Create registry service"
+  local kubectl="${KUBE_ROOT}/cluster/kubectl.sh"
+  "${kubectl}" create -f "${KUBE_ROOT}/cluster/addons/registry/registry-svc.yaml"
+}
+
+function wait-registry-readiness() {
+  if [[ "$ENABLE_CLUSTER_REGISTRY" == "true" ]]; then
```
Comment (roberthbailey) [inline cluster/libvirt-coreos/util.sh]: I'd prefer the "go-style" here where you check for != true and return and leave the rest of the function de-dented by 1 level.

File: cluster/libvirt-coreos/util.sh [hunk 3]
```diff
@@ -279,6 +285,51 @@ function kube-up {
...
+    echo "Wait for registry readiness..."
+    local kubectl="${KUBE_ROOT}/cluster/kubectl.sh"
```
Comment (roberthbailey) [inline cluster/libvirt-coreos/util.sh]: I thought we were moving away from referencing the shell script for kubectl. Is this still common in the rest of the libvirt_coreos shell?

File: cluster/libvirt-coreos/config-default.sh (hunk 1 above)
Comment (sdminonne) [inline cluster/libvirt-coreos/config-default.sh]: ok

File: cluster/libvirt-coreos/util.sh (hunk 2 above)
Comment (sdminonne) [inline cluster/libvirt-coreos/util.sh]: done

File: cluster/libvirt-coreos/util.sh (hunk 3 above)
Comment (sdminonne) [inline cluster/libvirt-coreos/util.sh]: sorry, it's something I was not aware of. I've modified the `util.sh` to use directly `kubectl` command that must be in the path (check added).
PTAL

**Comment by sdminonne** [comment]
Rebased. @roberthbailey thanks to have a look.

**Comment by roberthbailey** [comment]
please squash your commits, then lgtm

**Comment by sdminonne** [comment]
Thanks for the prompt reply. Squashed.

**Comment by sdminonne** [comment]
can I do this?
@k8s-bot gce etcd3 e2e test this

**Comment by roberthbailey** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by sdminonne** [comment]
@roberthbailey many thanks

**Pull request merged**